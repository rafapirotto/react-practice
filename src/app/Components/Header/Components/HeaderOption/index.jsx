import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import { DASHBOARD_ROUTE } from '../../../../routes';

const propTypes = {
  src: PropTypes.string.isRequired,
  styles: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  to: PropTypes.string,
  onClick: PropTypes.func,
};

const defaultProps = {
  to: DASHBOARD_ROUTE,
  onClick: () => {},
};

const HeaderOption = ({ src, alt, styles, to, onClick }) => (
  <Link to={to} onClick={onClick}>
    <img src={src} className={`mx-3 ${styles}`} alt={alt} />
  </Link>
);

HeaderOption.propTypes = propTypes;
HeaderOption.defaultProps = defaultProps;

export default HeaderOption;
