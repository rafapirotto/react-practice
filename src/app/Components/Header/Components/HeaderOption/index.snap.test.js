import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import HeaderOption from '.';
import search from '../../assets/search.png';

it('Matches the snapshot', () => {
  const component = renderer.create(
    <Router>
      <HeaderOption src={search} alt="search" styles="" />
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
