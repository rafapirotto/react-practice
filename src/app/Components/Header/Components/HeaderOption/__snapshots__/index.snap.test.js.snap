// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<a
  href="/dashboard"
  onClick={[Function]}
>
  <img
    alt="search"
    className="mx-3 "
    src="search.png"
  />
</a>
`;
